cmake/std/common.bash

```diff
@@ -0,0 +1,83 @@
+#!/usr/bin/env bash
+#
+# Common bash functions that we use in several scripts.
+#
+
+
+# Gets the current script name (full path + filename)
+function get_scriptname() {
+    # Get the full path to the current script
+    local script_name=`basename $0`
+    local script_path=$(dirname $(readlink -f $0))
+    local script_file="${script_path}/${script_name:?}"
+    echo "${script_file}"
+}
+
+
+# Gets the path to the current script (full path)
+function get_scriptpath() {
+    # Get the full path to the current script
+    local script_name=`basename $0`
```

Comment:
Don't think you need this line.

Reply:
`get_scriptname` and `get_scriptpath` were both in a different script, since I've added a `common.bash` script that holds some of these functions I moved them into it.

They're probably not necessary since other scripts are finding their locations and loading in the common.bash script via this:
```bash
SCRIPTFILE=$(realpath $BASH_SOURCE)
SCRIPTPATH=$(dirname $SCRIPTFILE)
source ${SCRIPTPATH:?}/common.bash
```

so we may not really need these functions any more but I don't think it hurts to hang onto them.  I'm not going to remove them for now.

I'm not going whole hog to optimize and clean up the bash stuff in this PR, I think that's an issue in itself.

Reply:
No problem.  I'm just asking questions as I run into them.  Filing the feedback away for later is fine with me.